lib/tools/android-manifest.js

```diff
@@ -144,7 +144,17 @@ async function extractApkInfoWithApkanalyzer (localApk, apkanalyzerPath) {
   if (!apkActivityAttribute) {
     throw new Error(`Cannot parse main activity name from ${manifestXml}`);
   }
-  const apkActivity = apkActivityAttribute.value;
+
+  let apkActivity = apkActivityAttribute.value;
+  // `android:name` has various cases from apkanalyzer.
+  // 1. ApiDemos
+  // 2. io.appium.mainactivity.MainTabActivity
+  // 3. .app.HelloWorld
+  // It should add package name only 1 case since 2 and 3 can be activity name properly.
+  if (!apkActivity.includes(".")) {
```

Comment:
I assume by doing this here we could also add an additional verification into what @dpgraham already changed in https://github.com/appium/appium-adb/pull/308/files and only push package/activity combination if the value does not start with package name already